### PR TITLE
Chatbot testing continued

### PR DESCRIPTION
**Pull request by rapala61**
#### What's this PR do?
It further strips in place middleware from the chatbot route and moves them to middleware modules for easier testing and troubleshooting.

#### How should this be reviewed?
- 👀 
- Run `npm run all-tests`
- Signup, reportback, deny broadcast, accept broadcast through Postman.

#### Any background context you want to provide?
Needs a little more extra testing love than the others.

#### Checklist
- [ ] Tested on Localhost.

### CODE REVIEW

File: app/routes/chatbot.js [hunk 1]
```diff
@@ -36,15 +35,15 @@ router.use(requiredParamsMiddleware(requiredParamsConf));
...
-router.use(createNewUserIfNotFoundMiddleware());
+router.use(
+  /**
+   * Check if DS User exists for given mobile number.
+   */
+  getUserMiddleware(),
+  /**
+   * Create DS User for given mobile number if we didn't find one.
+   */
+  createNewUserIfNotFoundMiddleware());
```
Comment (aaronschachter) [inline app/routes/chatbot.js]: I'm testing creating a new User locally, and can't figure out why I'm seeing this Retry message in the logs. Do you get the same @rapala61 ?
```
12:11:08 PM web.1 |  info: Receiving: incoming keyword
12:11:08 PM web.1 |  debug: contentful.fetchKeyword:luckybot
12:11:08 PM web.1 |  debug: contentful.fetchSingleEntry:{"content_type":"keyword","fields.keyword":"luckybot"}
12:11:08 PM web.1 |  info: /v1/chatbot post:{"incoming_image_url":"http://www.fox.com/sites/default/files/X-Files_hero.png","keyword":"luckybot"}
12:11:08 PM web.1 |  debug: User.lookup(mobile, 201706191211)
12:11:09 PM web.1 |  info: User.lookup could not find mobile:201706191211
12:11:09 PM web.1 |  debug: User.post
12:11:10 PM web.1 |  info: northstar.Users created user:594821cd10707d08a62063f3
12:11:10 PM web.1 |  info: This is a retry request. Retry count: 1
12:11:10 PM web.1 |  debug: phoenix.fetchCampaign:2900
```
I can replicate it by using a different new mobile number each time.

File: test/lib/helpers.test.js [hunk 2]
```diff
@@ -73,6 +78,28 @@ test.afterEach((t) => {
...
+  const error = { status: 500, message: 'error' };
+
+  // test
+  helpers.handlePhoenixPostError(t.context.req, t.context.res, error);
+  helpers.sendErrorResponse.should.have.been.calledWith(t.context.res, error);
+});
+
+test('handlePhoenixPostError should end conversation with error if API response is false', (t) => {
+  // setup
+  // TODO: This is quite brittle, this especific error should at least be in a config file, not
```
Comment (aaronschachter) [inline test/lib/helpers.test.js]: Great idea 👍

**Comment by aaronschachter** [comment]
I'm not sure why I'm getting these errors when running `npm run all-tests`, when wercker doesn't seem to be? I've made sure to `npm install`.

```
/Users/aschachter/Development/gambit/test/lib/dashbot.test.js
  12:22  error  Unable to resolve path to module 'nock'  import/no-unresolved

/Users/aschachter/Development/gambit/test/utils/factories/signup.js
  3:24  error  Unable to resolve path to module 'chance'  import/no-unresolved

/Users/aschachter/Development/gambit/test/utils/factories/user.js
  3:24  error  Unable to resolve path to module 'chance'  import/no-unresolved
```

File: app/routes/chatbot.js (hunk 1 above)
Comment (rapala61) [inline app/routes/chatbot.js]: Is it possible that you have this header set to 1: `x-blink-retry-count`?

**Comment by rapala61** [comment]
@aaronschachter try the good old `rm -rf node_modules/` and `npm i`

File: app/routes/chatbot.js (hunk 1 above)
Comment (aaronschachter) [inline app/routes/chatbot.js]: Completely possible!! :) sorry for false alarm!

**Pull request merged**